src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,18 +43,23 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query} // Привязываем значение к Redux
+          onChange={handleQueryChange} // Обновляем Redux при вводе текста
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
         <span className="icon is-right" style={{ pointerEvents: 'all' }}>
           {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
+          {query.length > 0 && (
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => clearSearch()}
```

Comment:
The `onClick` handler can be simplified by directly passing `clearSearch` without wrapping it in an arrow function. This is a minor optimization.